frame/glutton/src/weights.rs

```diff
@@ -0,0 +1,178 @@
+			.saturating_add(Weight::from_ref_time(209).saturating_mul(n.into()))
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	/// The range of component `n` is `[0, 1024]`.
+	fn waste_proof_size_some(n: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `762`
+		//  Estimated: `2499`
+		// Minimum execution time: 3_000 nanoseconds.
+		Weight::from_parts(6_773_699, 2499)
+			// Standard Error: 1_126
+			.saturating_add(Weight::from_ref_time(2_788).saturating_mul(n.into()))
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	/// The range of component `n` is `[0, 1024]`.
+	fn waste_proof_size_none(n: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `55`
+		//  Estimated: `2499`
+		// Minimum execution time: 3_000 nanoseconds.
+		Weight::from_parts(3_100_346, 2499)
+			// Standard Error: 341
+			.saturating_add(Weight::from_ref_time(683).saturating_mul(n.into()))
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+	}
+	/// Storage: Glutton Storage (r:1 w:0)
+	/// Proof: Glutton Storage (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton Compute (r:1 w:0)
+	/// Proof: Glutton Compute (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton TrashData (r:20 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	fn on_idle() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `12180`
+		//  Estimated: `50978`
+		// Minimum execution time: 78_503_000 nanoseconds.
+		Weight::from_parts(78_503_000_000, 50978)
+			.saturating_add(T::DbWeight::get().reads(22_u64))
+	}
+	fn empty_on_idle() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 1_000 nanoseconds.
+		Weight::from_ref_time(1_000_000)
+	}
+}
+
+// For backwards compatibility and tests
+impl WeightInfo for () {
+	/// The range of component `n` is `[0, 1024]`.
+	fn waste_ref_time(n: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 8_000 nanoseconds.
+		Weight::from_ref_time(7_942_950)
+			// Standard Error: 157
+			.saturating_add(Weight::from_ref_time(209).saturating_mul(n.into()))
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	/// The range of component `n` is `[0, 1024]`.
+	fn waste_proof_size_some(n: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `762`
+		//  Estimated: `2499`
+		// Minimum execution time: 3_000 nanoseconds.
+		Weight::from_parts(6_773_699, 2499)
```

Comment:
This consumes very little proof size, maybe we should do multiple reads? @ggwpez